crates/bevy_asset/src/server/mod.rs

```diff
@@ -940,11 +978,30 @@ impl AssetServer {
             .unwrap_or(RecursiveDependencyLoadState::NotLoaded)
     }
 
-    /// Returns true if the asset and all of its dependencies (recursive) have been loaded.
+    /// Convenience method that returns true if the asset has been loaded.
+    pub fn is_loaded(&self, id: impl Into<UntypedAssetId>) -> bool {
+        matches!(self.load_state(id), LoadState::Loaded)
+    }
+
+    /// Convenience method that returns true if the asset and all of its direct dependencies have been loaded.
     pub fn is_loaded_with_dependencies(&self, id: impl Into<UntypedAssetId>) -> bool {
```

Comment:
Am I seeing it right that current calls to `is_loaded_with_dependencies` now has different semantics and that users should replace it with `is_loaded_with_recursive_dependencies`? If so, this seems like an incredibly silent footgun. Users upgrading their version without looking at the migration guides will have no idea that their code is now slightly wrong. 
If my assessment is correct, we should ensure that this cannot happen, e.g. by leaving the semantics in place and introducing a `is_loaded_with_direct_dependencies`. I also don't like how the variant that you will usually want, namely `is_loaded_with_recursive_dependencies`, has a longer / more technical name than `is_loaded_with_dependencies`, which is probably not what you want.
But maybe I am misunderstanding the code?

Reply:
No misunderstanding, you are absolutely right. I will revert the name of `is_loaded_with_recursive_dependencies` to `is_loaded_with_dependencies` to preserve the original semantics. I will rename the check for self + direct deps to `is_loaded_with_direct_dependencies`